cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -115,6 +117,20 @@ public static LoggingContext getLoggingContext(Map<String, String> tags) {
                                        tags.get(ApplicationLoggingContext.TAG_RUN_ID),
                                        tags.get(ApplicationLoggingContext.TAG_INSTANCE_ID));
     } else if (tags.containsKey(WorkflowLoggingContext.TAG_WORKFLOW_ID)) {
+      if (tags.containsKey(WorkflowProgramLoggingContext.TAG_WORKFLOW_MAP_REDUCE_ID)) {
+        return new WorkflowProgramLoggingContext(namespaceId, applicationId,
+                                                 tags.get(WorkflowLoggingContext.TAG_WORKFLOW_ID),
+                                                 tags.get(ApplicationLoggingContext.TAG_RUN_ID), ProgramType.MAPREDUCE,
```

Comment:
i think we can just add the tag for node in metrics tags to use the program name from `WorkflowProgramLoggingContext`.